extension_service/datastore/providers/postgres.py

```diff
@@ -199,6 +199,48 @@ async def get_airport(self, id: int) -> Optional[models.Airport]:
         result = models.Airport.model_validate(dict(result))
         return result
 
+    async def get_airport_by_iata(self, iata: str) -> Optional[models.Airport]:
+        result = await self.__pool.fetchrow(
+            """
+              SELECT * FROM airports WHERE iata ILIKE $1
+            """,
+            iata,
+        )
+
+        if result is None:
+            return None
+
+        result = models.Airport.model_validate(dict(result))
+        return result
+
+    async def search_airports(
+        self,
+        country: Optional[str] = None,
+        city: Optional[str] = None,
+        name: Optional[str] = None,
+    ) -> list[models.Airport]:
+        if country is None:
+            country = "%"
+        if city is None:
+            city = "%"
+        name = "%" if name is None else f"%{name}%"
+
+        results = await self.__pool.fetch(
+            """
+            SELECT * FROM airports
+            WHERE (country IS NULL OR country ILIKE $1)
+            AND (city IS NULL OR city ILIKE $2)
+            AND (name IS NULL OR name ILIKE $3)
```

Comment:
I think this should be `($1 IS NULL OR country ILIKE $1)` -- essentially skip the statement if not provided. Then you can also remove `if country is None` section earlier.

Reply:
Updated! Thank youuu